drivers/sensor/adt7420/adt7420_trigger.c

```diff
@@ -132,24 +131,27 @@ int adt7420_init_interrupt(const struct device *dev)
 	const struct adt7420_dev_config *cfg = dev->config;
 	int rc;
 
-	drv_data->gpio = device_get_binding(cfg->int_name);
-	if (drv_data->gpio == NULL) {
-		LOG_DBG("Failed to get pointer to %s device!",
-			cfg->int_name);
-		return -EINVAL;
+	if (!device_is_ready(cfg->int_gpio.port)) {
+		if (cfg->int_gpio.port) {
+			LOG_ERR("%s: device %s is not ready", dev->name,
+						cfg->int_gpio.port->name);
+			return -ENODEV;
+		}
+
+		LOG_DBG("%s: int_gpio not defined in DT", dev->name);
+		return 0;
```

Comment:
Driver uses `GPIO_DT_SPEC_INST_GET`, not `GPIO_DT_SPEC_INST_GET_OR`, meaning `int_gpio.port` can't be NULL. We should also return `-ENODEV`.